wellcomecollection/catalogue-pipeline
# 6xx bottom up pt 2

## What does this change?

Part of https://github.com/wellcomecollection/catalogue-pipeline/issues/2563

This moves the (I think) final common fields - Subjects and Genres into the common MARC library.

## How to test

It may be prudent to run this on a new pipeline and compare with the existing one.  This has been a bit of a large piece of work, and as I have worked through it, I am not 100% confident that the existing Sierra test suite is without holes.  That said, I do believe this has preserved all relevant behaviour (save for one, see comment inline, where the conflicting behaviour was too awkward to accommodate)

## How can we measure success?

Genres and Subjects will now be extracted from MARC XML data.

## Have we considered potential risks?

Due to the intertwined nature of Subjects and Genres, along with the various inconsistencies in the way diverse Subjects were implemented, this is quite a large change.  I have already extracted and harmonised a lot of this content in previous PRs (https://github.com/wellcomecollection/catalogue-pipeline/pull/2620 and https://github.com/wellcomecollection/catalogue-pipeline/pull/2611), but this is where the "rubber hits the road", as it now implements the Sierra transformer in terms of the Marc transformer.

There is still potential to refactor this further, as well as taking decisions on conflicting behaviours.  I believe that I have isolated these sufficiently that once we choose one consistent way, we can make the changes easily
